------------------------------------------------------------
revno: 14071
revision-id: squid3@treenet.co.nz-20160817025501-e66sjxm0bfy3ksn3
parent: squid3@treenet.co.nz-20160805145933-0cpyy47o8955lamx
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=4428
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Wed 2016-08-17 14:55:01 +1200
message:
  Bug 4428: mal-formed Cache-Control:stale-if-error header
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20160817025501-e66sjxm0bfy3ksn3
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: b3b3ef13c45062a97bd5cc88c934019fe4af7a3c
# timestamp: 2016-08-17 02:55:20 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20160805145933-\
#   0cpyy47o8955lamx
# 
# Begin patch
=== modified file 'src/HttpHdrCc.cc'
--- src/HttpHdrCc.cc	2016-01-01 00:14:27 +0000
+++ src/HttpHdrCc.cc	2016-08-17 02:55:01 +0000
@@ -257,6 +257,27 @@
 
             /* for all options having values, "=value" after the name */
             switch (flag) {
+            case CC_BADHDR:
+                break;
+            case CC_PUBLIC:
+                break;
+            case CC_PRIVATE:
+                if (Private().size())
+                    packerPrintf(p, "=\"" SQUIDSTRINGPH "\"", SQUIDSTRINGPRINT(Private()));
+                break;
+
+            case CC_NO_CACHE:
+                if (noCache().size())
+                    packerPrintf(p, "=\"" SQUIDSTRINGPH "\"", SQUIDSTRINGPRINT(noCache()));
+                break;
+            case CC_NO_STORE:
+                break;
+            case CC_NO_TRANSFORM:
+                break;
+            case CC_MUST_REVALIDATE:
+                break;
+            case CC_PROXY_REVALIDATE:
+                break;
             case CC_MAX_AGE:
                 packerPrintf(p, "=%d", (int) maxAge());
                 break;
@@ -272,8 +293,14 @@
             case CC_MIN_FRESH:
                 packerPrintf(p, "=%d", (int) minFresh());
                 break;
-            default:
-                /* do nothing, directive was already printed */
+            case CC_ONLY_IF_CACHED:
+                break;
+            case CC_STALE_IF_ERROR:
+                packerPrintf(p, "=%d", staleIfError());
+                break;
+            case CC_OTHER:
+            case CC_ENUM_END:
+                // done below after the loop
                 break;
             }
 

